server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstraction.java

```diff
@@ -250,4 +252,42 @@ private boolean isNonEmpty(List<IndexMetaData> idxMetas) {
             return (Objects.isNull(idxMetas) || idxMetas.isEmpty()) == false;
         }
     }
+
+    class DataStream implements IndexAbstraction {
```

Comment:
Let's add the data stream index abstraction in another change? We can use the get index api in the yaml test to check that the index has correctly been created.

(the logic that adds data streams to the indicesLookup is missing here, which would need to be added too, but let's try to this pr small, so that it is easy to review)
(also a number of if statements in the code base may need to be revised because a new data stream type is introduced)